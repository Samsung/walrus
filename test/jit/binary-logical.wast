(module
(func (export "test1") (result i64 i64 i64) (local i64 i64)
  i64.const 0xff00ff00ff00ff00
  local.set 0
  i64.const 0x0ff00ff00ff00ff0
  local.set 1

  local.get 0
  i64.const 0x0ff00ff00ff00ff0
  i64.and

  i64.const 0xff00ff00ff00ff00
  local.get 1
  i64.or

  local.get 0
  local.get 1
  i64.xor

  (; 1080880403494997760, 18442521884633399280, 17361641481138401520 ;)
)

(func (export "test2") (param i32 i32) (result i32)
  block
    local.get 0
    local.get 1
    i32.and
    br_if 0
    i32.const 5
    return
  end

  i32.const 6
)

(func (export "test3") (param i32 i32) (result i32)
  i32.const 10
  i32.const 11

  local.get 0
  local.get 1
  i32.and
  select
)
)

(assert_return (invoke "test1") (i64.const 1080880403494997760) (i64.const 18442521884633399280) (i64.const 17361641481138401520))
(assert_return (invoke "test2" (i32.const 0x100) (i32.const 0x200)) (i32.const 5))
(assert_return (invoke "test2" (i32.const 0x0ff0) (i32.const 0x1f)) (i32.const 6))
(assert_return (invoke "test3" (i32.const 0xff) (i32.const 0xff00)) (i32.const 11))
(assert_return (invoke "test3" (i32.const 0x1ff) (i32.const 0xff00)) (i32.const 10))
