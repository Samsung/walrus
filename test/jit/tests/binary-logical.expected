[[[[[[[  Function   0  ]]]]]]]
Local 0: mapped to: 0 range [1 - 10]
Local 1: mapped to: 1 range [3 - 11]
Number of locals reduced from 2 to 2
0: Opcode: i64.const
  Result[0]: i64
1: Opcode: local.set
  Index: 0
  Param[0]: ref:0(0)
2: Opcode: i64.const
  Result[0]: i64
3: Opcode: local.set
  Index: 1
  Param[0]: ref:2(0)
4: Opcode: local.get
  Index: 0
  Result[0]: i64
5: Opcode: i64.const
  Result[0]: i64
6: Opcode: i64.and
  Param[0]: ref:4(0)
  Param[1]: ref:5(0)
  Result[0]: i64
7: Opcode: i64.const
  Result[0]: i64
8: Opcode: local.get
  Index: 1
  Result[0]: i64
9: Opcode: i64.or
  Param[0]: ref:7(0)
  Param[1]: ref:8(0)
  Result[0]: i64
10: Opcode: local.get
  Index: 0
  Result[0]: i64
11: Opcode: local.get
  Index: 1
  Result[0]: i64
12: Opcode: i64.xor
  Param[0]: ref:10(0)
  Param[1]: ref:11(0)
  Result[0]: i64
13: Opcode: return
  Param[0]: ref:6(0)
  Param[1]: ref:9(0)
  Param[2]: ref:12(0)
------------------------------
Frame size: 0, Arguments size: 48 (total: 48)
0: Opcode: i64.const
  Result[0]: i64 unused
1: Opcode: local.set
  Addr: sp[24]
  Param[0]: i64 const:0
2: Opcode: i64.const
  Result[0]: i64 unused
3: Opcode: local.set
  Addr: sp[32]
  Param[0]: i64 const:2
4: Opcode: local.get
  Addr: sp[24]
  Result[0]: i64 unused
5: Opcode: i64.const
  Result[0]: i64 unused
6: Opcode: i64.and
  Param[0]: i64 sp[24]
  Param[1]: i64 const:5
  Result[0]: i64 sp[0]
7: Opcode: i64.const
  Result[0]: i64 unused
8: Opcode: local.get
  Addr: sp[32]
  Result[0]: i64 unused
9: Opcode: i64.or
  Param[0]: i64 const:7
  Param[1]: i64 sp[32]
  Result[0]: i64 sp[8]
10: Opcode: local.get
  Addr: sp[24]
  Result[0]: i64 unused
11: Opcode: local.get
  Addr: sp[32]
  Result[0]: i64 unused
12: Opcode: i64.xor
  Param[0]: i64 sp[24]
  Param[1]: i64 sp[32]
  Result[0]: i64 sp[16]
13: Opcode: return
  Param[0]: i64 sp[0]
  Param[1]: i64 sp[8]
  Param[2]: i64 sp[16]

test1() => i64:1080880403494997760, i64:18442521884633399280, i64:17361641481138401520
